xml/System.Text.Json/JsonSerializer.xml

```diff
@@ -307,7 +307,7 @@ For more information, see [How to serialize and deserialize JSON](/dotnet/standa
 ]]></format>
         </remarks>
         <exception cref="T:System.IndexOutOfRangeException">
```

Comment:
Looking at the whole docs experience again, I don' think these version specific exceptions should be in this xml file.

We don't need (nor want) folks to see the Deserialize method can throw IOOR, **especially not on 3.1 and 5.0 (and beyond)**
That was a point-in-time bug in the code for a super uncommon edge case that folks shouldn't run into anyway.

This exception isn't something we want users to deal with, especially going forward.

**Please remove this exception all together.**

The docs are already versioned. A reader will ask, why am I seeing exceptions for the version I am not even using:
![image](https://user-images.githubusercontent.com/6527137/95504679-fa416700-0961-11eb-88dd-324025e84b96.png)

Reply:
Unfortunately the exceptions and remarks aren't versioned, only the API signature. So this is the way we document version-specific information. But I will remove it.